requirements/0003-audit-trail.md

```diff
@@ -0,0 +1,89 @@
+
+## Business Architecture
+
+### Capabilities
+
+Provide a way to track important state changes to the application's data, without the need to parse the logs.
+
+### Assumptions
+
+1. Functionality SHALL not be developed from scratch. An existing solution SHALL be used.
+
+### Dependencies
+
+There are available off-the-shelf FOSS solutions out there:
+- [pgMemento](https://github.com/pgMemento/pgMemento) - it is bare PostgreSQL solution using triggers. It is able to track  DDL changes, but it is not necessary and maybe even problematic given Alembic migrations. Stores audit data as JSONB deltas.
+- [SQLAlchemy-Continuum](https://github.com/kvesteri/SQLAlchemy-Continuum) - A proper SQLAlchemy extentions. Actively maintained. Supports Alembic migrations. Does not seem to support tracking WHO made a change. 78k downloads a month.
+- [Audit Trigger 91plus](https://github.com/2ndQuadrant/audit-trigger) - Does not track `SELECT` or `DDL` changes, but it is not required. Not sure how it'll work with Alembic migrations.
+- [PostgreSQL-Audit](https://github.com/kvesteri/postgresql-audit) - Seems like a next version of SQLAlchemy-Continuum. README claims it, and it's by the same developer. It has tho less downloads (6k vs 78k), less stars, less commits, and last change was done almost a year ago, compared to month ago in Continuum.
+
+Recommendation is to go with [SQLAlchemy-Continuum](https://github.com/kvesteri/SQLAlchemy-Continuum). Since it -- nor apparently neither of the plugins -- offers functionality of logging *who* made a change, development of custom plugin MAY be necessary.
+
+
+### Risks 
+
+### Business Risks
+
+Storing too much data in the separate audit table may create data breach issues, as well as make RODO-related activities like *right to be forgotten* more complex.
+
+### Impacts
+
+None.
+
+## Data Architecture
+
+A new table SHALL be created, tracking at least:
+- who made a change
+- when the change was made
+- which entity was modified
+- value before the change
+- value after the change
+
+## Application Architecture
+
+### Scenarios
+
+No impact.
+
+### User Interface
+
+No impact.
+
+### Services
+
+Endpoints SHALL track changes to Guest entity - including at least field `guest.claimed_by` and `guest.accommodation_unit_id`. 
+
+#### API
+
+No impact.
+
+### Availability
+
+No impact.
+
+### Security
+
+To mitigate the risk of data tampering of audit log, as well as data breach a separation of duties SHALL be implemented. The Services MUST NOT be able to read nor append to audit tables. Solution MAY use same schema in order to achieve that given appropriate access rights are put in place to ensure Services are not able to append -- willingly or via sql-injection type of attacks -- to the audit tables.
+
+## Infrastructure Architecture
+
+No impact.
```

Comment:
We'll need `iac` to define credentials for the non-superadmin app user that will be created by migrations (migrations will run using the current superadmin). `iac` will not be responsible for creating the account, only the coordination of creds between migrations and app.

We'll also need to stop providing the superadmin creds to the app and rotate those creds.